### PR TITLE
For #3443 - convert textPercentageSeekBar to Kotlin

### PR DESCRIPTION
**Pull request by nikit19**
### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by nikit19** [comment]
Hi @boek 
I was not able to reopen the pull request ( #3444 ) so I made new PR after resolving the conflicts.

File: app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt [hunk 1]
```diff
@@ -0,0 +1,484 @@
...
+    internal /* synthetic access */ var mSeekBarValue: Int = 0
+    internal /* synthetic access */ var mMin: Int = 0
+    private var mMax: Int = 0
+    private var mSeekBarIncrement: Int = 0
+    internal /* synthetic access */ var mTrackingTouch: Boolean = false
+    internal /* synthetic access */ var mSeekBar: SeekBar? = null
+    private var mSeekBarValueTextView: TextView? = null
+    private var mExampleTextTextView: TextView? = null
+    // Whether the SeekBar should respond to the left/right keys
+    /**
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt]: We probably shouldn't have three sets of KDoc including the getter, setter, and variable separately for each property. I recognize this is all from the original JavaDoc in the AOSP Java code.

File: app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt [hunk 2]
```diff
@@ -0,0 +1,484 @@
...
+     */
+    /**
+     * Sets whether the [SeekBar] should respond to the left/right keys.
+     *
+     * @param adjustable Whether the [SeekBar] should respond to the left/right keys
+     */
+    /* synthetic access */ var isAdjustable: Boolean = false
+    // Whether to show the SeekBar value TextView next to the bar
+    private var mShowSeekBarValue: Boolean = false
+    // Whether the SeekBarPreference should continuously save the Seekbar value while it is being
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt]: Same here. Three sets of KDoc is excessive.

File: app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt [hunk 3]
```diff
@@ -0,0 +1,484 @@
...
+            if (min > mMax) {
+                min = mMax
+            }
+            if (min != mMin) {
+                mMin = min
+                notifyChanged()
+            }
+        }
+
+    /**
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt]: Merge these into a single KDoc for the property.

File: app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt [hunk 4]
```diff
@@ -0,0 +1,484 @@
...
+    var seekBarIncrement: Int
+        get() = mSeekBarIncrement
+        set(seekBarIncrement) {
+            if (seekBarIncrement != mSeekBarIncrement) {
+                mSeekBarIncrement = Math.min(mMax - mMin, Math.abs(seekBarIncrement))
+                notifyChanged()
+            }
+        }
+
+    /**
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt]: Merge these into a property description.

File: app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt [hunk 5]
```diff
@@ -0,0 +1,484 @@
...
+            if (max < mMin) {
+                max = mMin
+            }
+            if (max != mMax) {
+                mMax = max
+                notifyChanged()
+            }
+        }
+
+    /**
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt]: Merge these.

File: app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt [hunk 6]
```diff
@@ -0,0 +1,484 @@
...
+     * @see .getShowSeekBarValue
+     */
+    var showSeekBarValue: Boolean
+        get() = mShowSeekBarValue
+        set(showSeekBarValue) {
+            mShowSeekBarValue = showSeekBarValue
+            notifyChanged()
+        }
+
+    /**
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/TextPercentageSeekBarPreference.kt]: Merge these.

**Comment by boek** [comment]
@nikit19 awesome, thank you! 🎉

**Comment by nikit19** [comment]
@colintheshots Thank you for the feedback. I have addressed the requested changes. If I missed something, please let me know.

**Comment by nikit19** [comment]
@colintheshots @boek Could you please review. I will happily make the changes if there are any changes to be done.

**Pull request merged**